pkg/k8s/watchers/service.go

```diff
@@ -434,6 +462,11 @@ func (k *K8sServiceWatcher) addK8sSVCs(svcID k8s.ServiceID, oldSvc, svc *k8s.Ser
 		logfields.K8sNamespace: svcID.Namespace,
 	})
 
+	if !option.Config.EnableServiceProtocolDifferentiation {
+		svc = stripServiceProtocol(svc)
```

Comment:
you have `oldSvc` and `svc`, wonder if you need to modify the `oldSvc` too

Reply:
yep, need to look better into this :+1:

Reply:
> wonder if you need to modify the oldSvc too

yes, if we want to compare apples to apples `oldSvc` needs to go through `stripServiceProtocol` in tandem with `svc` :+1: